src/components/scene/device-motion-permission-ui.js

```diff
@@ -0,0 +1,161 @@
+    deviceOrientation: {default: null},
+    deviceMotionEl: { default: '' }
+  },
+  init: function () {
+    this.deviceMotionEl = null;
+    this.onDeviceMotionClick = bind(this.onDeviceMotionClick, this);
+    this.onOrientationChangeClick = bind(this.onOrientationChangeClick, this);
+    this.grantedDeviceMotion = bind(this.grantedDeviceMotion, this);
+    if (typeof window.orientation !== 'undefined') {
+      try {
+        if (
+          DeviceOrientationEvent &&
+          typeof DeviceOrientationEvent.requestPermission === 'function'
+        ) {
+          DeviceOrientationEvent.requestPermission()
+            .then(response => {
+              if (response === 'granted') {
+                this.grantedDeviceMotion();
+              }
+            })
+            .catch(err => {
+              console.log(err);
+              this.deviceMotionEl = createDeviceMotionPermissionWindow(
+                this.onDeviceMotionClick,
+                this
+              );
+              this.el.appendChild(this.deviceMotionEl);
+            });
+        } else {
+          this.grantedDeviceMotion();
+        }
+      } catch (oops) {
+        this.grantedDeviceMotion();
+      }
+    } else {
+      this.remove();
+    }
+  },
+
+  remove: function () {
+    if (this.deviceMotionEl) {
+      this.el.removeChild(this.deviceMotionEl);
+    }
+  },
+
+  /**
+   * Enable device motion permission when clicked.
+   */
+  onDeviceMotionClick: function () {
+    try {
+      if (
+        DeviceOrientationEvent &&
+        typeof DeviceOrientationEvent.requestPermission === 'function'
+      ) {
+        DeviceOrientationEvent.requestPermission()
+          .then(response => {
+            if (response === 'granted') {
+              this.grantedDeviceMotion();
+            } else {
+              console.log('Device Motion permission not granted.');
+            }
+          })
+          .catch(console.error);
+      } else {
+        this.grantedDeviceMotion();
+      }
+    } catch (oops) {
+      console.log(
+        'Your device and application combination do not support device motion events.'
+      );
+    }
+  },
+
+  grantedDeviceMotion: function () {
+    const orientationChange=this.el.getAttribute('orientation-change');
+    const deviceOrientation=this.el.getAttribute('device-orientation');
+    this.remove();
+      if(orientationChange){
+        window.addEventListener('orientationchange', e => {
+          window[orientationChange](e);
+        });
+      }
+      if(deviceOrientation){
+        window.addEventListener('deviceorientation', e => {
+          window[deviceOrientation](e);
+        });
+      }
+    }
+  }
+);
+
+/**
+ * Create a button that when clicked will provide device motion permission.
+ *
+ * Structure: <div><button></div>
+ *
+ * @param {function} onClick - click event handler
+ * @returns {Element} Wrapper <div>.
+ */
+
+function createDeviceMotionPermissionWindow (onClick, obj) {
```

Comment:
Rename to createDeviceMotionPermissionDialog, update argument names and docs above. [Suggested changes](https://github.com/dmarcos/aframe/blob/b4be06e206821cedaa37a6b4512f4bd6641be1bd/src/components/scene/device-motion-permission-ui.js#L69)

Reply:
Done.